[FLOSS H2] Added documentation on how to run all tests with podman instead of docker (#4145)
ElektraInitiative/libelektra: doc/tutorials/run_all_tests_with_docker.md

## Prerequisites

> - Docker for Linux containers has to be preinstalled. Please refer to https://docs.docker.com/install/ if you haven't installed it yet. Your host OS can be either Linux, macOS or Windows of course. Alternatively, you can use podman which is a different container engine, which is developed by RedHat. See https://podman.io/ for more details and an installation guide.

Reviewer: ```suggestion
- Docker for Linux containers has to be pre-installed. Please refer to https://docs.docker.com/install/ if you haven't installed it yet. Your host OS can be either Linux, macOS or Windows. Alternatively, you can use podman, a different container engine, which is developed by RedHat. See https://podman.io/ for more details and an installation guide.
```

docker pull hub-public.libelektra.org/<image_name>:<tag_name>
```

> With podman:

Reviewer 2: ```suggestion
Or with podman:
```

```sh
podman pull hub-public.libelektra.org/<image_name>:<tag_name>
```

Example:

```sh
docker pull hub-public.libelektra.org/build-elektra-debian-stretch:201905-9dfe329fec01a6e40972ec4cc71874210f69933ab5f9e750a1c586fa011768ab
```

> With podman:

Reviewer 2: ```suggestion
Or with podman:
```

buildelektra-sid
```

#### 2.a Run the Docker Container with podman

> It is recommended that you run your container as a non-root user. As a result you have to set the correct permissions of the source directory.

Reviewer 3: ```suggestion
It is recommended that you run your container as a non-root user. As a result, you have to set the correct permissions of the source directory.
```

-rw-r--r--. 1 root root 10522  3. Nov 13:08 .cirrus.yml
-rw-r--r--. 1 root root  1479  3. Nov 13:08 .clang-format
-rw-r--r--. 1 root root  1621  3. Nov 13:08 .cmake-format.yaml
-rw-r--r--. 1 root root  3991  3. Nov 13:08 CMakeLists.txt
-rw-r--r--. 1 root root  3171  3. Nov 13:08 CODE_OF_CONDUCT.md
-rw-r--r--. 1 root root    25  3. Nov 13:08 .coveralls.yml
drwxr-xr-x. 1 root root   692  3. Nov 13:08 doc
drwxr-xr-x. 1 root root   888  3. Nov 13:08 examples
drwxr-xr-x. 1 root root   138  3. Nov 13:08 .git
drwxr-xr-x. 1 root root   148  3. Nov 13:08 .github
-rw-r--r--. 1 root root  1422  3. Nov 13:08 .gitignore
drwxr-xr-x. 1 root root   132  3. Nov 13:08 .idea
-rw-r--r--. 1 root root   558  3. Nov 13:08 lgtm.yml
-rw-r--r--. 1 root root  1530  3. Nov 13:08 LICENSE.md
drwxr-xr-x. 1 root root   134  3. Nov 13:08 LICENSES
-rw-r--r--. 1 root root   142  3. Nov 13:08 .oclint
-rw-r--r--. 1 root root  9714  3. Nov 13:08 README.md
-rw-r--r--. 1 root root   643  3. Nov 13:08 .restyled.yaml
drwxr-xr-x. 1 root root     8  3. Nov 13:08 .reuse
drwxr-xr-x. 1 root root   956  3. Nov 13:08 scripts
drwxr-xr-x. 1 root root   110  3. Nov 13:08 src
drwxr-xr-x. 1 root root   316  3. Nov 13:08 tests
```

> Inside the source directory you can change the permissions to any user id with `podman unshare chown 1000:1000 -R .`. Keep in mind that this

Reviewer 3: ```suggestion
Inside the source directory, you can change the permissions to any user id with `podman unshare chown 1000:1000 -R .`. Keep in mind that this
```
